From 1a04d0d6cb8630807cd57c6bbc15352076781ea5 Mon Sep 17 00:00:00 2001
Message-Id: <1a04d0d6cb8630807cd57c6bbc15352076781ea5.1413836944.git.chang-joon.lee@intel.com>
In-Reply-To: <5d8aa9994fff43965a6fab8c9a528b8b47a9d624.1413836944.git.chang-joon.lee@intel.com>
References: <5d8aa9994fff43965a6fab8c9a528b8b47a9d624.1413836944.git.chang-joon.lee@intel.com>
From: Deepak S <deepak.s@intel.com>
Date: Sun, 28 Sep 2014 13:05:15 +0530
Subject: [PATCH 71/71] REVERTME [VPG]: drm/i915: Cancel the hangcheck before
 runtime suspend

The cancel hangcheck timer was missed out in last forklift.
This causes a assert when we are in D0ix state.

This is a REVERTME because the upstream code is correct and this PATCH
should be discarded in next forklift.

Issue: GMINL-1918
Change-Id: Ie4170c2ccb1f1c9d01cb0d5bf3e920f96a0471e3
Signed-off-by: Deepak S <deepak.s@intel.com>
---
 drivers/gpu/drm/i915/i915_drv.c |    7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/i915_drv.c b/drivers/gpu/drm/i915/i915_drv.c
index 55ea272..35c44fd 100644
--- a/drivers/gpu/drm/i915/i915_drv.c
+++ b/drivers/gpu/drm/i915/i915_drv.c
@@ -1633,7 +1633,7 @@ static int intel_runtime_suspend(struct device *device)
 	struct pci_dev *pdev = to_pci_dev(device);
 	struct drm_device *dev = pci_get_drvdata(pdev);
 	struct drm_i915_private *dev_priv = dev->dev_private;
-	int ret;
+	int ret, i;
 
 	if (WARN_ON_ONCE(!(dev_priv->rps.enabled && intel_enable_rc6(dev))))
 		return -ENODEV;
@@ -1690,6 +1690,11 @@ static int intel_runtime_suspend(struct device *device)
 		return ret;
 	}
 
+	for (i = 0; i < I915_NUM_RINGS; i++) {
+		del_timer_sync(&dev_priv->ring[i].hangcheck.timer);
+		atomic_set(&dev_priv->ring[i].hangcheck.active, 0);
+	}
+
 	dev_priv->pm.suspended = true;
 
 	/*
-- 
1.7.9.5

